File: frontend/src/features/twin/types/deviceBaseline.ts
export type DeviceBaselineItem = {
  behavior_kind: string;
  behavior_key: string;
  count: number;
  first_seen_at: string;
  last_seen_at: string;
  established: boolean;
};

export type DeviceBaselineResponse = {
  device_id: string;
  profile_warm: boolean;
  total: number;
  items: DeviceBaselineItem[];
  suppress_count: number;
  suppress_age_hours: number;
  profile_min_keys: number;
};

File: backend/app/features/twin/schemas/device_baseline.py
from typing import List

from pydantic import BaseModel, Field


class DeviceBaselineItem(BaseModel):
    behavior_kind: str
    behavior_key: str
    count: int
    first_seen_at: str
    last_seen_at: str
    established: bool = False


class DeviceBaselineResponse(BaseModel):
    device_id: str
    profile_warm: bool = False
    total: int = 0
    items: List[DeviceBaselineItem] = Field(default_factory=list)
    suppress_count: int = 20
    suppress_age_hours: int = 0
    profile_min_keys: int = 5


class DeviceBaselineClearResponse(BaseModel):
    device_id: str
    cleared: int = 0
